src/api/ZoweExplorerZosmfApi.ts

```diff
@@ -29,24 +30,67 @@ class ZosmfApiCommon implements ZoweExplorerApi.ICommon {
     }
 
     private session: Session;
+    private log: Logger;
     constructor(public profile?: IProfileLoaded) {
     }
 
     public getProfileTypeName(): string {
         return ZosmfUssApi.getProfileTypeName();
     }
 
+    public getSessionFromCommandArgument(cmdArgs: ICommandArguments): Session{
+        return zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
+    }
+
     public getSession(profile?: IProfileLoaded): Session {
         if (!this.session) {
-            this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
+            try {
+                if (!this.profile.profile.tokenValue) {
+                    this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
+                } else {
+                    const serviceProfile = this.profile;
+                    const cmdArgs: ICommandArguments = {
+                        $0: "zowe",
+                        _: [""],
+                        host: serviceProfile.profile.host,
+                        port: serviceProfile.profile.port,
+                        basePath: serviceProfile.profile.basePath,
+                        rejectUnauthorized: serviceProfile.profile.rejectUnauthorized !== null,
+                        tokenType: serviceProfile.profile.tokenType,
+                        tokenValue: serviceProfile.profile.tokenValue
+                    };
+
+                    this.session = zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
+                }
+            } catch (error) {
+                this.log.debug(error);
+            }
         }
         return this.session;
     }
 
     public async getStatus(validateProfile?: IProfileLoaded, profileType?: string): Promise<string> {
         // This API call is specific for z/OSMF profiles
+        let validateSession: Session;
         if (profileType === "zosmf") {
-            const validateSession = await zowe.ZosmfSession.createBasicZosmfSession(validateProfile.profile);
+            if (validateProfile.profile.tokenValue) {
+                const serviceProfile = validateProfile;
+                const cmdArgs: ICommandArguments = {
+                    $0: "zowe",
+                    _: [""],
+                    host: serviceProfile.profile.host,
+                    port: serviceProfile.profile.port,
+                    basePath: serviceProfile.profile.basePath,
+                    rejectUnauthorized: serviceProfile.profile.rejectUnauthorized !== null,
+                    tokenType: serviceProfile.profile.tokenType,
+                    tokenValue: serviceProfile.profile.tokenValue
+                };
+
+                validateSession = zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
```

Comment:
this line 89 should also use the call `this.getSessionFromCommandArgument(cmdArgs)`

Reply:
changed it

Reply:
in fact to reduce the amount of duplicate code with the gathering of ICommandArguments line 78-92 could just call `this.getSession(validateProfile.profile);` and just have it return the session here.

Reply:
it looks like the same code in both so this should do the same as line 76-92 right?
```
if (profileType === "zosmf") {
  validateSession = this.getStatus(validateProfile);
}
```